core/src/main/java/nu/yona/server/analysis/entities/IntervalActivity.java

```diff
@@ -29,10 +62,53 @@ public Goal getGoal()
 		return goal;
 	}
 
+	public LocalDate getDate()
+	{
+		return date;
+	}
+
 	public ZonedDateTime getStartTime()
 	{
 		return startTime;
 	}
 
+	public boolean areAggregatesComputed()
+	{
+		return aggregatesComputed;
+	}
+
 	public abstract ZonedDateTime getEndTime();
+
+	public abstract ChronoUnit getTimeUnit();
+
+	public List<Integer> getSpread()
+	{
+		if (areAggregatesComputed())
+		{
+			return spread;
+		}
+
+		return computeSpread();
+	}
+
+	protected abstract List<Integer> computeSpread();
+
+	public int getTotalActivityDurationMinutes()
+	{
+		if (areAggregatesComputed())
+		{
+			return totalActivityDurationMinutes;
+		}
+
+		return computeTotalActivityDurationMinutes();
+	}
+
+	protected abstract int computeTotalActivityDurationMinutes();
+
+	protected static List<Integer> getEmptySpread()
+	{
+		return Arrays.asList(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
```

Comment:
That's why I wrote it in this way: I thought you would pick it out at review and give a better suggestion :)